spec/models/user_spec.rb

```diff
@@ -1,5 +1,37 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'Validations' do
+    subject { User.new(name: 'KJC', bio: 'Hello Internet! I am KJC.') }
+
+    before { subject.save }
+
+    it 'Name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'Bio should be present' do
+      subject.bio = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'User should have post greater than or equal to 0' do
+      subject.posts_counter = -1
+      expect(subject).to_not be_valid
+    end
+
+    it 'User should have post greater than or equal to 0' do
+      subject.posts_counter = 0
+      expect(subject).to be_valid
+    end
+  end
+
+  describe 'Should test recent post method' do
+    before { 4.times { |post_nb| Post.create(user_id: subject.id, title: "This is post #{post_nb}") } }
+
+    it 'User should have three recent posts' do
+      expect(subject.recent_posts).to eq(subject.posts.last(3))
+    end
+  end
 end
```

Comment:
- [ ] Currently, all your tests are not passing, I have attached the image of the error I received when I ran your tests. After debugging the error, I found out that your tables were not created properly. Your tables do not have the required columns/attributes needed for each table according to the [ERD diagram](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_app_erd.png). Kindly create all the attributes needed in each table according to the ERD diagram so that your tests can pass.

Your schema | Error message
:---:|:---:
![](https://i.imgur.com/dyAQB9S.png) | ![](https://i.imgur.com/o9BLOyL.png)

Reply:
Hi @julie-ify 
This is strange, because on my side all the tests work properly. So I don't understand what's the problem on your side.
What could be the reason for that?

Reply:
Hi @Kingjosh007,
 Do you have a different schema from the one I posted? Kindly verify as I reviewed the project based on what I have in your branch and I got the errors I posted. I can't really say but if you think it's an error kindly request another review but I strongly ask that you take a look at your migrations okay.

Reply:
Kindly let me know if you need my help okay.

Reply:
@julie-ify 
I just check my migration files, and I don't know what happened, but couldn't see any column definition there!
I had them in the previous milestones. Anyway, I just added it back and migrated again, but this Rails thing is so confusing...

Reply:
@julie-ify 
Should I request another review now?

Reply:
@Kingjosh007,

If you have added the migrations and tested it and it's passing then you can request another review I guess

Reply:
@julie-ify 
Thank you so much.